tests/clients/test_conversation.py

```diff
@@ -0,0 +1,734 @@
+                name='get_weather',
+                description='Get weather information for a location',
+                parameters={
+                    'type': 'object',
+                    'properties': {
+                        'location': {
+                            'type': 'string',
+                            'description': 'The city and state, e.g. San Francisco, CA',
+                        },
+                        'unit': {
+                            'type': 'string',
+                            'enum': ['celsius', 'fahrenheit'],
+                            'description': 'Temperature unit',
+                        },
+                    },
+                    'required': ['location'],
+                },
+            )
+        )
+
+    def create_calculate_tool(self):
+        """Create a calculate tool for testing."""
+        return ConversationTools(
+            function=ConversationToolsFunction(
+                name='calculate',
+                description='Perform mathematical calculations',
+                parameters={
+                    'type': 'object',
+                    'properties': {
+                        'expression': {
+                            'type': 'string',
+                            'description': 'Mathematical expression to evaluate',
+                        }
+                    },
+                    'required': ['expression'],
+                },
+            )
+        )
+
+    def create_user_message(self, text):
+        """Helper to create a user message for Alpha2."""
+        return ConversationMessage(
+            of_user=ConversationMessageOfUser(content=[ConversationMessageContent(text=text)])
+        )
+
+    def create_system_message(self, text):
+        """Helper to create a system message for Alpha2."""
+        return ConversationMessage(
+            of_system=ConversationMessageOfSystem(content=[ConversationMessageContent(text=text)])
+        )
+
+    def create_assistant_message(self, text, tool_calls=None):
+        """Helper to create an assistant message for Alpha2."""
+        return ConversationMessage(
+            of_assistant=ConversationMessageOfAssistant(
+                content=[ConversationMessageContent(text=text)], tool_calls=tool_calls or []
+            )
+        )
+
+    def create_tool_message(self, tool_id, name, content):
+        """Helper to create a tool message for Alpha2."""
+        return ConversationMessage(
+            of_tool=ConversationMessageOfTool(
+                tool_id=tool_id, name=name, content=[ConversationMessageContent(text=content)]
+            )
+        )
+
+
+class ConversationAlpha1SyncTests(ConversationTestBase, unittest.TestCase):
+    """Synchronous Alpha1 conversation API tests."""
+
+    def test_basic_conversation_alpha1(self):
+        """Test basic Alpha1 conversation functionality."""
+        with DaprClient(f'{self.scheme}localhost:{self.grpc_port}') as client:
```

Comment:
This line is in every single test, I think it's worth simplifying.
For example we could add a setup step where we do `self.client = DaprClient(f'{self.scheme}localhost:{self.grpc_port}')` and then use `self.client`.
WDYT?